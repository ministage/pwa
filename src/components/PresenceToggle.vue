<template>
  <v-switch
      class="big-toggle ml-3 pt-0"
      v-model="value"
      inset
      color="#6fcf97"
      error-count="0"
      hide-details
      :loading="loading"
  ></v-switch>
</template>

<script>
export default {
  name: "PresenceToggle",
  props: {
    enabled: Boolean,
    loading: Boolean,
    onToggle: Function
  },
  computed: {
    value: {
      get: function(){
        return this.enabled;
      },
      set: function(newValue){
        this.onToggle(newValue);
      }
    },
  },
}
</script>

<style lang="scss">
.big-toggle {
  margin-top: 0 !important;
  transform: scale(2);
  transform-origin: left;
}
</style>