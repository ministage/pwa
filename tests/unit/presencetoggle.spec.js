// Libraries
import Vue from 'vue'
import Vuetify from 'vuetify'

// Components
import PresenceToggle from '@/components/PresenceToggle'

// Utilities
import {createLocalVue, mount} from '@vue/test-utils'

describe('PresenceToggle.vue', () => {
    const localVue = createLocalVue()
    let vuetify

    beforeEach(() => {
        vuetify = new Vuetify()
    })

    it('should display aanwezig when enabled', () => {
        const wrapper = mount(PresenceToggle, {
            localVue,
            vuetify,
            propsData: {enabled: true},
        })

        // We could also verify this differently
        // by checking the text content
        const title = wrapper.find('div > span')

        expect(title.text()).toBe('Aanwezig')
    })

    it('should display afwezig when disabled', () => {
        const wrapper = mount(PresenceToggle, {
            localVue,
            vuetify,
            propsData: {enabled: false},
        })

        // We could also verify this differently
        // by checking the text content
        const title = wrapper.find('div > span')

        expect(title.text()).toBe('Afwezig')
    })

    it('should be green when enabled', function () {
        const wrapper = mount(PresenceToggle, {
            localVue,
            vuetify,
            propsData: {enabled: true},
        });
        const color = wrapper.find('div')
        expect(color.classes()).toContain('on')
    })

    it('should be red when disabled', function () {
        const wrapper = mount(PresenceToggle, {
            localVue,
            vuetify,
            propsData: {enabled: false},
        });
        const color = wrapper.find('div')
        expect(color.classes()).toContain('off')
    })

    it('should execute the callback when clicked', () => {
        const event = jest.fn()
        const wrapper = mount(PresenceToggle, {
            localVue,
            vuetify,
            propsData: { enabled: true, onToggle: event },
        })

        const button = wrapper.find('div')

        expect(event).toHaveBeenCalledTimes(0)

        // Simulate a click on the toggle
        button.trigger('click')

        // Ensure that our mock callback was called
        expect(event).toHaveBeenCalledTimes(1)

        // Ensure that the right data was given
        expect(event).toHaveBeenCalledWith(true)
    })
});