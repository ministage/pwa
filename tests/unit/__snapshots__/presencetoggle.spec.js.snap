// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresenceToggle.vue should have a custom title and match snapshot 1`] = `
<div class="rounded-pill flex flex-row flex-grow-0 align-center on"><span class="ml-3 mr-2">Aanwezig</span>
  <div class="h-12 w-12 mt-1 mb-1 mr-1" style="background-color: white; border-radius: 50%;"></div>
  <!---->
</div>
`;
